todo_app/data/controller.py

```diff
@@ -0,0 +1,192 @@
+        self._name = value
+        return self._name
+
+    @property
+    def id(self):
+        return self._id
+
+    @id.setter
+    def id(self, value: str):
+        self._id = value
+        return self._id
+
+    @property
+    def idShort(self):
+        return self._idShort
+
+    @idShort.setter
+    def idShort(self, value: str):
+        self._idShort = value
+        return self._idShort
+
+    @property
+    def description(self):
+        return self._description
+
+    @description.setter
+    def description(self, value: str):
+        self._description = value
+        if not self._description:
+            self._description = "No description available"
+        return self._description
+
+    @property
+    def idBoard(self):
+        return self._idBoard
+
+    @idBoard.setter
+    def idBoard(self, value: str):
+        self._idBoard = value
+        return self._idBoard
+
+    @property
+    def listName(self):
+        return self._listName
+
+    @listName.setter
+    def listName(self, value: str):
+        self._listName = value
+        if not self._listName:
+            self._listName = "No Name Found"
+        return self._listName
+
+    @property
+    def taskUrl(self):
+        return self._taskUrl
+
+    @taskUrl.setter
+    def taskUrl(self, value):
+        self._taskUrl = value
+        return self._taskUrl
+
+
+def getItems(documentList: pymongo.cursor.Cursor) -> list[object]:
+
+    """
+    Gets a list of documents as a
+
+    Args:
+        documentList: a pymongo Cursor from the database
+
+    Returns:
+        cardList: a list of Card objects with attributes added.
+    """
+    cardList = []
+
+    for document in documentList:
+        card = Card()
+        card.listName = document["status"]
+        card.name = document["title"]
+        card.description = document["description"]
+        card.id = str(document["_id"]).strip("'")
+        card.idShort = card.id[:7]
+        cardList.append(card)
+
+    return cardList
+
+
+def getItem(collection: pymongo.cursor.Cursor, id: str) -> object:
+    """
+    Fetches the MongoDB document with the specified ID.
+
+    Args:
+        id: The ID of the item.
+        collection: the MongoDB document collection object
+
+    Returns:
+        card: the card representation of the MongoDB document,
+        or None if no items match the specified ID.
+    """
+
+    returnedDocument = collection.find_one({"_id": ObjectId(id)})
+
+    try:
+        card = Card()
+        card.id = returnedDocument["_id"]
+        card.description = returnedDocument["description"]
+        card.name = returnedDocument["title"]
+        card.listName = returnedDocument["status"]
+
+        return card
+
+    except:
+        return None
+
+
+def updateTask(
+    collection: pymongo.cursor.Cursor, id, status: str
+) -> pymongo.results.UpdateResult:
+
+    update = collection.update_one(
+        {"_id": ObjectId(f"{id}")}, {"$set": {"status": f"{status}"}}
```

Comment:
There's not much point in using an f string to convert an existing string into a new one!
```suggestion
        {"_id": ObjectId(id)}, {"$set": {"status": status}}
```